zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/handlers/UserTaskHandlerTest.java

```diff
@@ -48,10 +53,29 @@ public class UserTaskHandlerTest {
           UserTaskIntent.MIGRATED,
           UserTaskIntent.ASSIGNED,
           UserTaskIntent.UPDATED);
+
   private final ProtocolFactory factory = new ProtocolFactory();
   private final String indexName = "test-tasklist-task";
   private final TestFormCache formCache = new TestFormCache();
-  private final UserTaskHandler underTest = new UserTaskHandler(indexName, formCache);
+  private final ExporterMetadata exporterMetadata = new ExporterMetadata();
+  private final UserTaskHandler underTest =
+      new UserTaskHandler(indexName, formCache, exporterMetadata);
+
+  @BeforeEach
+  void resetMetadata() {
+    exporterMetadata.setFirstUserTaskKey(TaskImplementation.ZEEBE_USER_TASK, -1);
+  }
+
+  private static Stream<Arguments> taskIdArguments() {
+    return Stream.of(
+        // First 8.7 UserTask key, Record Key, PI key, FNI key, expected id
+        Arguments.of(100, 110, 111, 211, 211), Arguments.of(100, 10, 111, 211, 10));
+  }
```

Comment:
👍🏼 

I think if we extract the logic into a helper, we could test this even more narrowed. Just as a thought. With two tests that are even named what they actually verify/test